frontend/src/components/MatchingPairs/MatchingPairs.js

```diff
@@ -18,75 +18,66 @@ const MatchingPairs = ({
-            case -10: return '-10 <br />Misremembered';
-            case 0: return '0 <br />No match';
-            case 10: return '+10 <br />Lucky match';
-            case 20: return '+20 <br />Good job!';
-            default: return '';
-        }
-    }
-
     const registerUserClicks = (posX, posY) => {
         xPosition.current = posX;
         yPosition.current = posY;
     }
 
     // Show (animated) feedback after second click on second card or finished playing
-    const showFeedback = () => {
+    const showFeedback = (score) => {
 
         const turnedCards = sections.filter(s => s.turned);
         // Check if this turn has finished
         if (turnedCards.length === 2) {
             // update total score & display current score
-            setTotal(total + score.current);
-            setMessage(setScoreMessage(score.current));
-            setMessage(setScoreMessage(score.current));
-            // show end of turn animations if enabled
-            if (showAnimation) {
-                switch (score.current) {
-                    case 10:
-                        turnedCards[0].lucky = true;
-                        turnedCards[1].lucky = true;
-                        break;
-                    case 20:
-                        turnedCards[0].memory = true;
-                        turnedCards[1].memory = true;
-                        break;
-                    default:
-                        turnedCards[0].nomatch = true;
-                        turnedCards[1].nomatch = true;
-                        // reset nomatch cards for coming turns
-                        setTimeout(() => {
-                            turnedCards[0].nomatch = false;
-                            turnedCards[1].nomatch = false;
-                        }, 700);
-                        break;
-                }
+            setTotal(total + score);
+            let fbclass;
+            switch (score) {
+                case 10:
+                    fbclass = 'fblucky';
+                    setFeedbackText('Lucky match');
+                    break;
+                case 20:
+                    fbclass = 'fbmemory';
+                    setFeedbackText('Good job!');
+                    break;
+                case 0:
+                    fbclass = 'fbnomatch';
+                    setFeedbackText('No match');
+                    break;
+                case -10:
+                    fbclass = 'fbmisremembered';
+                    setFeedbackText('Misremembered');
+                    break;
+                default:
+                    setFeedbackClass('');
+                    setFeedbackText('');
             }
-
-
-            // add third click event to finish the turn
-            document.getElementById('root').addEventListener('click', finishTurn);
+            setFeedbackClass(fbclass);
+            turnedCards[0].matchClass = turnedCards[1].matchClass = fbclass;
+            turnedCards[1].seen = turnedCards[1].seen = true;
```

Comment:
I see you are setting properties directly on objects in an array that are defined within this function. Does that actually do something? The values of the properties are probably lost after the execution of the function / after a re-render, or not?

Reply:
Good point, I should check that. From manual testing, it seems like this works. The matchClass is something that will only be used for an animation, so loosing that information wouldn't be problematic. `seen`, however, would.